# Project 8 -- [Drew] [Caldwell]

**TA Help:** N/A

    
**Collaboration:** ChatGPT
    
- Used to help with syntax errrors and how to create the first couple of functions.

## Question 1

In [3]:
# Load the necessary data
library(data.table)
death_records <- fread("/anvil/projects/tdm/data/death_records/DeathRecords.csv")

In [4]:
# Define the makeatable function
makeatable <- function(x) {
  prop.table(table(x, useNA="always"))
}

In [5]:
# Display the table for the Sex column
sex_table <- makeatable(death_records$Sex)
print(sex_table)

x
        F         M      <NA> 
0.4939664 0.5060336 0.0000000 


The death records set is almost exactly 50/50 for male and female

In [6]:
# Display the table for the MaritalStatus column
marital_status_table <- makeatable(death_records$MaritalStatus)
print(marital_status_table)

x
          D           M           S           U           W        <NA> 
0.152388043 0.372463819 0.126575962 0.007112043 0.341460133 0.000000000 


## Question 2

In [7]:
# Define the teenagecount function
teenagecount <- function(x) {
  length(x[(x >= 13) & (x <= 19) & (!is.na(x))])
}

In [8]:
# Count teenagers in DeathRecords
teenagers_death_records <- teenagecount(death_records$Age)
print(teenagers_death_records)

[1] 12643


We have 12,643 teenager deaths in the records

In [9]:
# Load the Olympics data
library(data.table)
olympics_data <- fread("/anvil/projects/tdm/data/olympics/athlete_events.csv")

In [10]:
# Count teenagers in the Olympics data
teenagers_olympics <- teenagecount(olympics_data$Age)
print(teenagers_olympics)

[1] 32250


32,250 teenagers have competed in the Olympics (for this data set)

## Question 3

In [5]:
# Load the beer reviews data\
library(data.table)
beer_reviews <- fread("/anvil/projects/tdm/data/beer/reviews_sample.csv")

In [6]:
# Define the longesttest function
longesttest <- function(x) {
  x[which.max(nchar(x))]
}

In [7]:
# Find the longest review in beer reviews
longest_review <- longesttest(beer_reviews$text)
print(longest_review)

[1] "\302\240\302\240 32 fl oz brown glass Alesmith grumbler filled earlier today at The Growler Room in Austin, Texas, and stored in my fridge until consumption. Served into a Spaten stangekrug. Reviewed at low altitude live here in Austin on 08/31/13. Cost was $ $21 USD. Expectations are sky high given both its current ratings and the description \342\200\223 which makes it sound tasty. This is my first Texan sour. Served frigid cold and allowed to warm following the initial pour. Side-poured with extra vigor as slight carbonation issues are anticipated given that it\342\200\231s a growler pour. A: No bubble show forms as I pour. Pours a one finger wide head of gorgeous khaki colour. Great creaminess \342\200\223 reminiscent actually of an Irish Dry Stout\342\200\231s head. Good thickness. Supple. Has an appealing soft complexion. Okay lacing; some legs cling to the sides of the glass when I tilt it, but they quickly fall. Retention is pretty good \342\200\223 about 3-4 minutes; not 

In [8]:
# Load the election data
library(data.table)
election_data <- fread("/anvil/projects/tdm/data/election/itcont1980.txt", quote="")
names(election_data) <- c("CMTE_ID", "AMNDT_IND", "RPT_TP", "TRANSACTION_PGI", "IMAGE_NUM", "TRANSACTION_TP", "ENTITY_TP", "NAME", "CITY", "STATE", "ZIP_CODE", "EMPLOYER", "OCCUPATION", "TRANSACTION_DT", "TRANSACTION_AMT", "OTHER_ID", "TRAN_ID", "FILE_NUM", "MEMO_CD", "MEMO_TEXT", "SUB_ID")

In [9]:
# Find the longest name in the election data
longest_name <- longesttest(election_data$NAME)
print(longest_name)

[1] "REPUBLICAN NATIONAL COMMITTEE - CONTRIBUTIONS (AKA REPUBLICAN NATIONAL FINANCE COMMITTEE"


The longest name in the electiondata was "REPUBLICAN NATIONAL COMMITTEE - CONTRIBUTIONS (AKA REPUBLICAN NATIONAL FINANCE COMMITTEE"

## Question 4

In [10]:
# Define the mostpopulardate function
mostpopulardate <- function(date_column) {
  most_popular <- as.character(names(which.max(table(date_column))))
  count <- max(table(date_column))
  return(list(date = most_popular, count = count))
}

In [11]:
# Find the most popular date in beer reviews
popular_date_reviews <- mostpopulardate(beer_reviews$date)
print(popular_date_reviews)

$date
[1] "2011-11-15"

$count
[1] 1497



Most popular date to give a beer review from the given data was 11/15/2011 where 1497 reviews were written

In [12]:
# Find the most popular transaction date in the election data
popular_date_election <- mostpopulardate(election_data$TRANSACTION_DT)
print(popular_date_election)


$date
[1] "12311979"

$count
[1] 2562



Most Popular election date was 12/31/1979 and the count of it was 2562

## Question 5

In [2]:
# Load the flight data
library(data.table)
flights_data <- fread("/anvil/projects/tdm/data/flights/subset/1990.csv")

In [3]:
# Define the myaveragedelay function
myaveragedelay <- function(airport_code) {
  delays <- flights_data$DepDelay[flights_data$Origin == airport_code & !is.na(flights_data$DepDelay)]
  mean_delay <- mean(delays, na.rm = TRUE)
  return(mean_delay)
}

In [4]:
# Calculate average departure delays for Indianapolis
avg_delay_ind <- myaveragedelay("IND")
print(avg_delay_ind)

[1] 5.969772


Average delayed flight at the Indy Airport is 5.96 hours

In [5]:
# Calculate average departure delays for New York City
avg_delay_jfk <- myaveragedelay("JFK")
print(avg_delay_jfk)

[1] 11.85727


Averaged delayed flight at JFK is 11.85 hours

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.